prisma/schema.prisma

```diff
@@ -0,0 +1,48 @@
+generator client {
+  provider        = "prisma-client-js"
+  previewFeatures = ["fullTextSearch", "fullTextIndex"]
+}
+
+datasource db {
+  provider = "mysql"
+  url      = env("DATABASE_URL")
+}
+
+model user {
+  email      String    @id @db.VarChar(50)
+  username   String    @unique(map: "username") @db.VarChar(50)
+  first_name String    @db.VarChar(50)
+  last_name  String    @db.VarChar(50)
+  password   String    @db.Char(60)
+  role       String    @default("user") @db.VarChar(80)
+  user_role  user_role @relation(fields: [role], references: [id], onUpdate: Restrict, map: "user_ibfk_1")
```

Comment:
what is the diff of role and user_role?

Reply:
I'm actually not sure. What I did was create the user and user_role tables on mysql cli and use prisma db pull to generate the prisma schema. Will look into what line 18 means. But right now I think it establishes a one to many relation with the user_role table. Will look into what map means.

Reply:
I looked it up in the prisma docs and i got.

onUpdate: Restrict - prevents the identifier of a referenced record from being changed.
In the context of the user_role and user tables, this means that user_roles id cannot be changed.

i can't find map: in the doccs, let me try to remove it and see what happens

Reply:
map: was using the user_ibkf_1 index for some reason. I suspect it's something that present on my tables since I created the tables initially using mysql cli and then did prisma db pull. I removed map in that relation and just used db push to make the tables according to the prisma schema.

Reply:
Hi Rico, I don't think we have to worry about the map. How does the fields `role` and `user_role` relate to each other? I'm trying to understand hehe

Reply:
What is the use of user_role table?

Reply:
I wanted to set the role column of the user table to be a foreign key referencing the id column of the user_role table to establish a constraint. This would prevent the back end code from adding users with roles not present in the table. But now that I think about it, user_role table could actually be removed. Should I just remove user_role table?